packages/fxa-settings/src/pages/ResetPassword/AccountRecoveryConfirmKey/index.test.tsx

```diff
@@ -4,47 +4,46 @@
 
 import React from 'react';
 import '@testing-library/jest-dom/extend-expect';
-import { fireEvent, render, screen } from '@testing-library/react';
+import { fireEvent, render, screen, waitFor } from '@testing-library/react';
 // import { getFtlBundle, testAllL10n } from 'fxa-react/lib/test-utils';
 // import { FluentBundle } from '@fluent/bundle';
-import { usePageViewEvent, logViewEvent } from '../../../lib/metrics';
+import { logPageViewEvent, logViewEvent } from '../../../lib/metrics';
 import { viewName } from '.';
 import {
   Subject,
   MOCK_RECOVERY_KEY,
   MOCK_RESET_TOKEN,
   MOCK_RECOVERY_KEY_ID,
   MOCK_KB,
+  mockCompleteResetPasswordParams,
+  paramsWithMissingToken,
+  paramsWithMissingCode,
+  paramsWithMissingEmail,
 } from './mocks';
 import { REACT_ENTRYPOINT } from '../../../constants';
 import { Account } from '../../../models';
 import { getSearchWithParams, typeByLabelText } from '../../../lib/test-utils';
 import { AuthUiErrors } from '../../../lib/auth-errors/auth-errors';
-import { act } from 'react-dom/test-utils';
+import { MOCK_ACCOUNT } from '../../../models/mocks';
 
 jest.mock('../../../lib/metrics', () => ({
-  usePageViewEvent: jest.fn(),
+  logPageViewEvent: jest.fn(),
   logViewEvent: jest.fn(),
 }));
-
-type ParamValue = string | null;
-
-let account: Account;
-let mockToken: ParamValue,
-  mockCode: ParamValue,
-  mockEmail: ParamValue,
-  mockUid: ParamValue;
 const mockNavigate = jest.fn();
 
+const mockSearchParams = {
+  mockEmail: mockCompleteResetPasswordParams.email,
+  mockPasswordHash: mockCompleteResetPasswordParams.emailToHashWith,
+  mockToken: mockCompleteResetPasswordParams.token,
+  mockCode: mockCompleteResetPasswordParams.code,
+  mockUid: mockCompleteResetPasswordParams.uid,
+};
+
 const mockLocation = () => {
-  const search = getSearchWithParams({
-    mockToken,
-    mockCode,
-    mockEmail,
-    mockUid,
-  });
+  const search = new URLSearchParams(getSearchWithParams(mockSearchParams));
```

Comment:
Nice, I thought we wouldn't need `getSearchWithParams` since I'd nixed it in `CompleteResetPassword` tests when introducing `LinkValidator` but looks like we can still use it like this. Maybe I need to do this [here](https://github.com/mozilla/fxa/pull/15392/files#r1213632830). 🤷‍♀️

Reply:
We sure do have a variety of patterns that we'll want to clean up. `getSearchParams` was useful for mocking here 👍

Reply:
Actually, I realized we don't need this. Will nix it in the next commit. With an update to `mockSearchParams` to match the keys to the expected search params (e.g., 'email' instead of 'mockEmail'), we can directly call `new URLSearchParams(mockSearchParams)`